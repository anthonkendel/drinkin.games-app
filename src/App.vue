<template>
  <div id="app">
    <div v-if="!authentication && !noAuthentication">
      <div class="container">
        <SignUp v-if="signUp" />
        <SignIn v-else />
      </div>
    </div>
    <div v-else-if="authentication || noAuthentication">
      <Navbar />
      <div class="container">
        <div class="columns">
          <div class="column is-2 is-hidden-touch">
            <SidebarGames />
          </div>
          <div class="column">
            <section class="section">
              <EmailVerificationWarning />
              <router-view :key="$route.fullPath" />
            </section>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import Navbar from '@/components/Navbar';
import SidebarGames from '@/components/SidebarGames';
import SignIn from '@/components/SignIn';
import SignUp from '@/components/SignUp';
import EmailVerificationWarning from '@/components/EmailVerificationWarning';
import { mapState } from 'vuex';

export default {
  name: 'DrinkinGamesApp',
  components: { SignUp, SignIn, Navbar, SidebarGames, EmailVerificationWarning },
  computed: {
    ...mapState({
      signUp: 'signUp',
      authentication: 'authentication',
      noAuthentication: 'noAuthentication',
    }),
  },
};
</script>

<style lang="scss">
@import url('https://fonts.googleapis.com/css?family=Proza+Libre|Roboto|Roboto+Mono');
@import '~bulma/sass/utilities/_all';

$family-sans-serif: 'Roboto', sans-serif;
$family-monospace: 'Roboto Mono', monospace;

.title,
.subtitle {
  font-family: 'Proza Libre', sans-serif;
}

$size-1: 2.7rem;
$size-2: 2.2rem;
$size-3: 1.7rem;
$size-4: 1.2rem;
$size-5: 1.15rem;
$size-6: 0.85rem;
$size-7: 0.7rem;

$section-padding: 1rem 0.75rem;

$radius-small: 2px;
$radius: 3px;
$radius-large: 4px;

$speed: 100ms;

$primary: hsl(48, 100%, 68%);
$primary-invert: findColorInvert($primary);

$info: hsl(200, 100%, 68%);
$info-invert: findColorInvert($info);

$success: hsl(150, 100%, 68%);
$success-invert: findColorInvert($success);

$warning: hsl(30, 100%, 68%);
$warning-invert: findColorInvert($warning);

$danger: hsl(10, 100%, 68%);
$danger-invert: findColorInvert($danger);

// Setup $colors to use as bulma classes (e.g. 'is-twitter')
$colors: (
  'white': (
    $white,
    $black,
  ),
  'black': (
    $black,
    $white,
  ),
  'light': (
    $light,
    $light-invert,
  ),
  'dark': (
    $dark,
    $dark-invert,
  ),
  'primary': (
    $primary,
    $primary-invert,
  ),
  'info': (
    $info,
    $info-invert,
  ),
  'success': (
    $success,
    $success-invert,
  ),
  'warning': (
    $warning,
    $warning-invert,
  ),
  'danger': (
    $danger,
    $danger-invert,
  ),
);

@import '~bulma';
@import '~buefy/src/scss/buefy';
</style>
