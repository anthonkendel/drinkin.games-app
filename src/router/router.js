import Home from '@/components/Home.vue';
import Details from '@/components/Details.vue';
import Vue from 'vue';
import Router from 'vue-router';

Vue.use(Router);

const router = new Router({
  mode: 'history',
  base: process.env.BASE_URL,
  routes: [
    {
      path: '/',
      name: 'Home',
      component: Home,
    },
    {
      path: '/:id',
      name: 'Details',
      props: true,
      component: Details,
    },
  ],
});

export default router;
