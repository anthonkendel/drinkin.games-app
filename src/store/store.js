import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    signUp: false,
    authentication: false,
    noAuthentication: false,
  },
  mutations: {
    setSignUp: (state, signUp) => (state.signUp = signUp),
    setAuthentication: (state, authentication) => (state.authentication = authentication),
    setNoAuthentication: (state, noAuthentication) => (state.noAuthentication = noAuthentication),
  },
  actions: {
    setSignUp: ({ commit }, signUp) => commit('setSignUp', signUp),
    setAuthentication: ({ commit }, authentication) => commit('setAuthentication', authentication),
    setNoAuthentication: ({ commit }, noAuthentication) => commit('setNoAuthentication', noAuthentication),
  },
});
