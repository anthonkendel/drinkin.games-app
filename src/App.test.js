import App from '@/App';
import router from '@/router/router';
import store from '@/store/store';
import { mount } from '@vue/test-utils';

describe('App.vue', () => {
  it('should match snapshot', () => {
    const wrapper = mount(App, {
      router,
      store,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
