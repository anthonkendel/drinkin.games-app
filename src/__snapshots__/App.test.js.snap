// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should match snapshot 1`] = `
<div
  id="app"
>
  <div>
    <section
      class="section"
    >
      <div
        class="section"
      >
        <div
          class="columns is-centered"
        >
          <div
            class="column is-2"
          >
            <h2
              class="title"
            >
              Sign in
            </h2>
             
            <form>
              <div
                class="field"
              >
                <label
                  class="label"
                  for="email"
                >
                  
              Email
            
                </label>
                 
                <p
                  class="control has-icons-right"
                >
                  <input
                    class="input"
                    id="email"
                    placeholder="Email"
                    required="required"
                    type="email"
                  />
                   
                  <span
                    class="icon is-right"
                  >
                    <i
                      class="fas fa-user"
                    />
                  </span>
                </p>
              </div>
               
              <div
                class="field"
              >
                <label
                  class="label"
                  for="password"
                >
                  
              Password
            
                </label>
                 
                <p
                  class="control has-icons-right"
                >
                  <input
                    class="input"
                    id="password"
                    placeholder="Password"
                    required="required"
                    type="password"
                  />
                   
                  <span
                    class="icon is-right"
                  >
                    <i
                      class="fas fa-lock"
                    />
                  </span>
                </p>
              </div>
               
              <div
                class="field is-grouped"
              >
                <p
                  class="control is-expanded"
                >
                  <button
                    class="button is-primary is-fullwidth"
                  >
                    Sign in
                  </button>
                </p>
                 
                <p
                  class="control is-expanded"
                >
                  <button
                    class="button is-text is-fullwidth"
                    type="button"
                  >
                    
                Sign up
              
                  </button>
                </p>
              </div>
            </form>
          </div>
        </div>
         
        <div
          class="columns is-centered"
        >
          <div
            class="column is-2"
          >
            <p>
              
          You don't want to create an account?
          Click 
              <a>
                here
              </a>
               to continue without an account.
        
            </p>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
